### ISSUE TITLE
Add the ability for clients to upload the backups themselves

### ISSUE THREAD

**Issue by brainkiller** [backuppc/backuppc#367]
In my case I'm using backuppc to backup pods running in a kubernetes cluster.
For now I have to configure and run an rsync/ftp/... server for every container I want to backup.
That while many of the containers have rsync or ftp client added by default.

It would be great if somehow backuppc could offer the ability to have the clients upload the backup themselves.
That way I could eliminate the rsync containers and ftp servers.

I guess it should be possible to add another option to the xfer tab to add rsync-server or ftp-server.
And to distinguish the different clients based on their login credentials..
A write only account would make sense from a security perspective.

I assume the backup upload could be jailed to the pc/<client>/<latest> folder.

The above is just the rough idea, I have no idea on how to start on it or if it's the best way to integrate this into the backuppc architecture

**Comment by craigbarratt** [comment]
Yes, this would be a nice feature.  However, opening up in-bound ssh to the server adds risk.  Perhaps the only safe way to allow client-initiated backups would be via rsyncd, and make it write-only as you suggest.  Also, the client would have to figure whether to do a full/incremental, which for example could be manually set via cron.

This is a potential long-term feature request that I will think about some more.

**Comment by GWHAYWOOD** [comment]
Closed - unlikely that sufficient development effort will be available in the foreseeable future.
